### PR TITLE
build: ignore macOS make deploy artefacts & add them to clean-local

### PR DESCRIPTION
**Pull request by fanquake**

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK, the file is produced in the makefile (though, not by any umbrella rule), so it should be ignored

also code review ACK 290c5c6ae80668a1e8090be74f653b0f24704d97

**Comment by practicalswift** [comment]
utACK 57fb62ec23c8d78a761577b87dc38090afcd13f3

File: .gitignore [hunk 1]
```diff
@@ -128,3 +128,8 @@ db4/
 
 # clang-check
 *.plist
+
+osx_volname
+dist/
```
Comment (jonasschnelli) [inline .gitignore]: Do you have any other files than the already ignored `Bitcoin-Qt.app` in here?

**Comment by jonasschnelli** [comment]
ACK 57fb62ec23c8d78a761577b87dc38090afcd13f3

**Comment by hebasto** [comment]
@fanquake 
Testing `make deploy` on High Sierra 10.13.6. Cannot observe mentioned artefacts.
Which macOS version do you use?

**Comment by fanquake** [comment]
@hebasto The artefacts occur when building for a macOS HOST on a Linux machine.

File: .gitignore [hunk 2]
```diff
@@ -128,3 +128,8 @@ db4/
...
+dpi36.background.tiff
```
Comment (hebasto) [inline .gitignore]: Could be just `*.background.tiff`?

**Comment by hebasto** [comment]
Tested master branch on Linux Mint 19.2:
```
cd depends
make HOST=x86_64-apple-darwin14 SDK_PATH=`pwd`/SDKs
cd ..
./configure --prefix=`pwd`/depends/x86_64-apple-darwin14
make deploy
```

Artefacts:
- `dist/` folder -- observed
- `dpi36.background.tiff` -- observed
- `dpi72.background.tiff` -- observed
- `osx_volname` -- _not_ observed

**Comment by fanquake** [comment]
@hebasto Thanks for testing. I've fixed your nit.

> osx_volname -- not observed

You should see it if you run `make osx_volname`

File: .gitignore (hunk 2 above)
Comment (fanquake) [reply .gitignore]: Done.

**Comment by hebasto** [comment]
ACK c84ff23c6daade5145d960e1ac26596ddede2da1, tested on Linux Mint 19.2:
- `make` with targets `osx_volname`, `deploydir` and `deploy` creates the mentioned artefacts
- `make clean` removes them

**Comment by dongcarl** [comment]
ACK c84ff23c6daade5145d960e1ac26596ddede2da1

**Pull request merged**